application/src/main/kotlin/net/corda/v5/application/persistence/PersistenceService.kt

```diff
@@ -110,44 +101,110 @@ interface PersistenceService {
     fun <T : Any> findAll(entityClass: Class<T>): PagedQuery<T>
 
     /**
-     * Create a [ParameterisedQuery] to support a named query to return a list of entities of the given type in a single transaction. Casts results to the specified type [T].
+     * Create a [ParameterisedQuery] to support a named query to return a list of entities of the given type in a single
+     * transaction. Casts result set to the specified type [T].
      * Example usage:
-     * ```java
-     * // For JPA Entity:
-     * @CordaSerializable
-     * @Entity
-     * @Table(name="DOGS")
-     * @NamedQuery(name="find_by_name_and_age", query="SELECT d FROM Dog d WHERE d.name = :name AND d.age <= :maxAge")
-     * public class Dog {
-     *  @Id
-     *  private UUID id;
-     *  @Column(name="DOG_NAME", length=50, nullable=false, unique=false)
-     *  private String name;
-     *  @Column(name="DOG_AGE")
-     *  private Int age;
-     *
-     * // getters and setters
-     * ...
+     *
+     * - Kotlin:
+     *
+     * ```kotlin
+     * class MyFlow : RPCStartableFlow {
```

Comment:
I think its easier to understand without the inclusion of the flow in the example.

Should we include a very small example in the interfaces kdoc rather than for `query`?

Reply:
Isn't it also a bad example  as I thought we'd guide people to put their JPA classes in the contract CPK, not the flow?